### ISSUE TITLE
web-mode can't recognize engine correctly for Org-mode babel buffer

### ISSUE THREAD

**Issue by stardiviner** [fxbois/web-mode#678]
I set org-mode Babel to open rhtml block with web-mode.

```lisp
(add-to-list 'org-src-lang-modes '("rhtml" . web))
```

And I have configure company-robe for completing for Rails code in `erb` engine (I get `erb` with `(web-mode-language-at-pos)`).

```lisp
(defadvice company-robe (before web-mode-set-up-ac-sources activate)
    "Set `robe-mode' based on current language before running `company-robe'."
    (if (equal major-mode 'web-mode)
        (let ((web-mode-cur-language (web-mode-language-at-pos)))
          (if (string= web-mode-cur-language "erb")
              (unless robe-mode (robe-mode))
            (if robe-mode (robe-mode -1))
            ))))
```

The upper advice work correctly in normal file `index.html.erb` file.

But web-mode can't recognize engine correctly in Org-mode babel block.

Here is a test Org-mode file content:

```org
#+BEGIN_SRC rhtml
<%= button_to "An article", @article, remote: true %>
#+END_SRC
```

When I run `(web-mode-language-at-pos)`, it shows `html`.

**Comment by fxbois** [comment]
if you want to test the template engine, use ```web-mode-engine```

**Comment by stardiviner** [comment]
But in Org-mode babel buffer, the variable `web-mode-engine` is `"none"`. Still can work in my case.

**Comment by stardiviner** [comment]
I guess the problem is on how `web-mode` determine current buffer's engine (like `erb` etc).
Because org-mode babel buffer don't have filename? so failed?

**Comment by fxbois** [comment]
You would have to force the engine ... there is a function for that : ```web-mode-set-engine```

**Comment by stardiviner** [comment]
I found the solution,
```lisp
;; [C-c ']
  ;; let Org-mode Babel src code block auto set `web-mode-engine' for rhtml.
  (defadvice org-edit-special (before org-edit-src-code activate)
    (let ((lang (nth 0 (org-babel-get-src-block-info))))
      (if (string= lang "rhtml")
          (web-mode-set-engine "erb"))
      ))
```

Can you add support for this by a macro which can support all web-mode supported engines? That would be very useful!

**Comment by fxbois** [comment]
can you propose this macro ?

**Comment by stardiviner** [comment]
@fxbois How can I get the key-value pair for the modes and engines? I can take a try to write a macro to iterate over all engines.

**Comment by fxbois** [comment]
look at ```web-mode-engines```

**Comment by stardiviner** [comment]
@fxbois I tried, but failed. I don't know ther major-modes associate with engines. I only know `rhtml-mode` for `erb`. And I found might this don't need a macro, just a function is enough.
I also found a bug in my upper advice code. `(web-mode-set-engine "erb")` is setting engine in org buffer, instead of opened src block babel buffer. So when I open babel block like `#+BEGIN_SRC emacs-lisp`, still get `erb` with `(web-mode-detect-engin)`.
